[new] Add InteractiveDataDisplay w/ git auto-update

ElenaPochernina opened the pull request:
Pull request for issue: #
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/predictionmachines/InteractiveDataDisplay
 * Official website (optional, not the repository):
 * NPM package url (optional):
 * License and its reference: Apache-2.0 (https://github.com/predictionmachines/InteractiveDataDisplay/blob/master/LICENSE)
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 14
   - Count of stars: 33
   - Count of forks: 18
 * NPM download stats (optional):
   - Downloads in the last day:
   - Downloads in the last week:
   - Downloads in the last month:

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [ ] No already exist / duplicated issue and PR
 * [ ] The lib has notable popularity
   * [ ] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [ ] Close corresponding issue in commit message
 * [ ] Mention related issue(s), people in commit message, comment.

pvnr0082t commented on ajax/libs/InteractiveDataDisplay/package.json:
```diff
@@ -0,0 +1,30 @@
+    {       
+      "basePath": "dist",
+      "files": [
+        "**/*"       
```
Because cdnjs doesn't support `*.d.ts` file, let's use `"**/*.!(d.ts)"`. Thanks.

ElenaPochernina commented:
@pvnr0082t I add `author` and `homepage` info in package.json. Also now I use `"**/*.!(d.ts)"`, but build is still failed.

pvnr0082t commented:
@ElenaPochernina CI test failed because the branch is too old. Would you please rebase it onto the latest master branch? If there is any problem, feel free to ask and we will help you. :+1:

ElenaPochernina commented:
@pvnr0082t Thanks. I have two questions:
1. Can I rebase the branch onto the latest master branch, but not downloading the whole repository?
2. I read that the library must have required popularity, but we haven't got this. But this library is used in many other projects (https://notebooks.azure.com/api/libraries/fsharp/html/FSharp%20for%20Azure%20Notebooks.ipynb#Other-Charting-Libraries,-e.g.-Angara.Chart
https://blogs.msdn.microsoft.com/visualstudio/2016/12/05/azure-notebooks-now-support-f/
http://biomodelanalyzer.org/tool.html
http://fetchclimate2.cloudapp.net/)
Is it possible to publish this library without required popularity or not?

pvnr0082t commented:
@ElenaPochernina 
> 1. Can I rebase the branch onto the latest master branch, but not downloading the whole repository?

You can try to use sparseCheckout feature. Take a look at [sparseCheckout.md](https://github.com/cdnjs/cdnjs/blob/master/documents/sparseCheckout.md)
> 2. Is it possible to publish this library without required popularity or not?

Maybe we should ask CDNJS maintainer. @PeterDaveHello Can we host this library?

PeterDaveHello commented:
If the author would like to help us review and confirm this PR, okay, I'll accept this one, but that's not a common case, should better confirm before send a new PR first.

ElenaPochernina commented:
@pvnr0082t I rebased the branch onto the latest master branch. And also I fixed `package.json`, so now all checks have passed.

PeterDaveHello commented:
@ElenaPochernina Great! Thanks!

pvnr0082t commented on ajax/libs/interactive-data-display/package.json:
```diff
@@ -0,0 +1,35 @@
+      {
+        "basePath": "dist",
+        "files": [
+          "**/*.!(d.ts)"
```
@ElenaPochernina Sorry for my mistake. This config will also fetch `".d.ts"` files. Let's modify to `"**/*.!(*ts)"`. Thanks.

maruilian11 commented:
@ElenaPochernina there are two ways to add this lib:
1. only add package.json without `version` field
2. add package.json with `"version": "1.5.0",`  and all its real files under dist in `1.5.0`.

I think remove the `version` field in package.json and `1.5.0/idd.min.js` would be the simpler way.

pvnr0082t commented on ajax/libs/interactive-data-display/package.json:
```diff
@@ -0,0 +1,34 @@
+  "name": "interactive-data-display",
+  "filename": "idd.min.js",
+  "description": "A JavaScript visualization library for dynamic data",
+  "homepage": "https://github.com/predictionmachines/InteractiveDataDisplay/wiki",
```
@ElenaPochernina I think it's better to use https://predictionmachines.github.io/InteractiveDataDisplay/ as `homepage`.

ElenaPochernina commented:
@maruilian11 I removed the `version` field in package.json and also `1.5.0/idd.min.js`.
@pvnr0082t I changed the field `homepage` in package.json on ` https://predictionmachines.github.io/InteractiveDataDisplay/`.

Also I rebased the branch onto the latest master branch.
